lib/OpenQA/Worker/Job.pm

```diff
@@ -568,6 +569,20 @@ sub _format_reason {
 sub _set_job_done {
     my ($self, $reason, $params, $callback) = @_;
 
+    # Retrieve extended reason if available
+    my $pooldir    = $self->worker->pool_directory;
+    my $state_file = path($pooldir)->child(BASE_STATEFILE);
+    try {
+        if (-e $state_file) {
+            my $msg = decode_json($state_file->slurp)->{msg};
+            $reason = "$reason: " . $msg unless $msg =~ /\n/;
+        }
+    }
+    catch {
+        $reason = "$reason: Corrupted state file could not be read";
+        log_warning("Found $state_file but failed to parse the JSON: $_");
+    };
+
```

Comment:
I did and we do. See also the unit test covering that.

Reply:
@okurz This is not "defensive programming". The error is *fully/appropriately* handled at the closest level with enough context to do so. Besides, letting the exception fall though here would break everything here.